src/AppCard.test.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
```

Comment:
This tests in this file work, but shouldn't we be testing from the perspective of the user and not be reaching into the internals which the user cannot directly react? 

Do you agree with this: https://testing-library.com/docs/guiding-principles

Reply:
I think those are good principles for testing UI, but in the world where you're testing network protocol code, I think unit tests on key functions of the code can be very useful.

Reply:
and to be clear, functional and integration tests that show the overall effect are super useful. I don't meant to take away from them. It's just hard to have them cover every use case, especially the error cases, without making tests pretty unwieldy.